create_coda_files/create_coda_files.py

```diff
@@ -1,55 +1,153 @@
-    parser.add_argument("variable_name", metavar="variable-name",
-                        help="Name of message variable in flow")
-    parser.add_argument("coda_output_path", metavar="coda-output-path",
-                        help="Path to a Coda file to write processed messages to")
-    parser.add_argument("prev_coda_path", metavar="prev-coda-path",
-                        help="Path to a Coda file containing previously coded messages")
+# opens a coding scheme
+def _open_scheme(filepath):
+    with open(filepath, "r") as f:
+        firebase_map = json.load(f)
+        return Scheme.from_firebase_map(firebase_map)
+
+#Autocleaners that exist for this project
+AUTO_CLEANERS ={'gender':somali.DemographicCleaner.clean_gender,
+                'age':somali.DemographicCleaner.clean_age,
+                'yes_no':somali.DemographicCleaner.clean_yes_no,
+                }
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        description='Takes a TracedData json file, cleans the data and'
+        'converts it to a CODAv2 file for verfication and manual coding')
+    parser.add_argument(
+        'user', help='User launching this program')
+    parser.add_argument(
+        'json_input_path', metavar='json-input-path',
+        help='Path to the input JSON file, containing a list of serialized'
+        'TracedData objects')
+    parser.add_argument(
+        'flow_name', metavar='flow-name',
+        help='Name of activation flow from which this data was derived')
+    parser.add_argument(
+        'variable_name', metavar='variable-name',
+        help='Name of message variable in flow')
+    parser.add_argument(
+        'traced_json_output_path', metavar='json-output-path',
+        help='Path to concatenated TraceData JSON')
+    parser.add_argument(
+        'coda_output_path', metavar='coda-output-path',
+        help='Path to a Coda file to write processed messages to')
+    parser.add_argument(
+        'coding_scheme_path', metavar='coding-scheme',
+        help='Path to coding scheme that is for this data')
+    parser.add_argument(
+        'is_yes_no', metavar='is-yes-no',
+        help='Is this variable a yes no question')
+    parser.add_argument(
+        'auto_cleaner', metavar='auto-cleaner',
+        help='automated cleaner to apply to this data')
 
     args = parser.parse_args()
     user = args.user
     json_input_path = args.json_input_path
     flow_name = args.flow_name
     variable_name = args.variable_name
+    traced_json_output_path = args.traced_json_output_path
     coda_output_path = args.coda_output_path
-    if not os.path.exists(prev_coda_path):
-        prev_coda_path = None
+    coding_scheme_path = args.coding_scheme_path
+    is_yes_no = args.is_yes_no
+    if args.auto_cleaner in AUTO_CLEANERS.keys():
+        auto_cleaner = args.auto_cleaner
+        auto_cleaner = AUTO_CLEANERS[auto_cleaner]
```

Comment:
This is a bit weird. Why not just to do auto_cleaner = AUTO_CLEANERS[args.auto_cleaner]?

Reply:
Cleaned up to clearup the options